tools/accuracy_checker/custom_evaluators/colorization_v2_evaluator.py

```diff
@@ -0,0 +1,215 @@
+
+        output_blob = next(iter(self.exec_network.outputs))
+        (h_orig, w_orig) = input_data[0].shape[:2]
+
+        new_result = []
+        for input in input_data:
+            input = input.astype(np.float32)
+            img_lab = cv2.cvtColor(input, cv2.COLOR_RGB2Lab)
+            img_l = img_lab[:, :, 0]
+
+            img_lab_rs = cv2.cvtColor(input, cv2.COLOR_RGB2Lab)
+            img_l_rs = img_lab_rs[:, :, 0]
+            img_l_rs -= self.test_mean
+
+            res = self.exec_network.infer(inputs={self.input_blob: [img_l_rs]})
+            (n_out, c_out, h_out, w_out) = res[output_blob].shape
+            update_res = np.zeros((n_out, 2, h_out, w_out)).astype(np.float32)
+            color_coeff = np.load(self.color_coeff)
+
+            for res_blob, color_coeff_blob in list(zip(res[output_blob][0, :, :, :], color_coeff)):
+                for upd_res_blob, clr_coeff in list(zip(update_res[0, :, :, :], color_coeff_blob)):
+                    upd_res_blob += res_blob * clr_coeff
+
+            out = update_res[0, :, :, :].transpose((1, 2, 0))
+            out = cv2.resize(out, (w_orig, h_orig))
+            img_lab_out = np.concatenate((img_l[:, :, np.newaxis], out), axis=2)
+            self.img_bgr_out = np.clip(cv2.cvtColor(img_lab_out, cv2.COLOR_Lab2BGR), 0, 1)
+            new_result.append(self.img_bgr_out)
+        return new_result
+
+    def release(self):
+        del self.exec_network
+        del self.img_bgr_out
+
+    def fit_to_input(self, input_data):
+        input_data = np.reshape(input_data, self.network.inputs[self.input_blob].shape)
+        return {self.input_blob: input_data}
+
+
+class ColorizationCheckModel(BaseModel):
+    def __init__(self, network_info, preprocessing_info, launcher):
+        super().__init__(network_info, launcher)
+        model_xml = str(network_info['checker']['model'])
+        model_bin = str(network_info['checker']['weights'])
+
+        self.network = launcher.create_ie_network(model_xml, model_bin)
+        if hasattr(launcher, 'plugin'):
+            self.exec_network = launcher.plugin.load(self.network)
+        else:
+            launcher.load_network(self.network)
+            self.exec_network = launcher.exec_network
+        self.input_blob = next(iter(self.network.inputs))
+        self.output_blob = next(iter(self.network.outputs))
+        self.adapter = create_adapter(network_info['checker']['adapter'])
+        self.adapter.output_blob = self.output_blob
+        self.data_std = float(([elem for elem in preprocessing_info
+                                if elem['type'] == 'normalization'][0]['std']))
+
+    def predict(self, identifiers, input_data):
+        result = self.exec_network.infer(self.fit_to_input(input_data))
+        result = self.adapter.process([result], identifiers, [{}])
+        return result
+
+    def release(self):
+        del self.exec_network
+
+    def fit_to_input(self, input_data):
+        input_data *= self.data_std
```

Comment:
I think better to handle std on output colorization network and are you sure that it can be different from 255?

Reply:
@eaidova done)